app/src/components/market/sections/market_buy/scalar_market_buy.tsx

```diff
@@ -166,11 +173,16 @@ export const ScalarMarketBuy = (props: Props) => {
       )
       const pricesAfterTrade = MarketMakerService.getActualPrice(balanceAfterTrade)
 
-      const newPrediction = pricesAfterTrade[1] * ((upperBound || 0) - (lowerBound || 0)) + (lowerBound || 0)
+      const newPrediction = calcPrediction(
+        pricesAfterTrade[1].toString(),
+        scalarLow || new BigNumber(0),
+        scalarHigh || new BigNumber(0),
+        18,
```

Comment:
Yeah, so the difficulty here is that outcome shares are always 18 decimals, but I can see it being difficult to track which places should use 18 decimals, and which should use `collateral.decimals`. To remedy this, I was thinking it might be good to use a constant value to represent 18 decimals, at least for clarity. What do you think? Know of any better way?